### PR TITLE
Add AKS support in mvp_demo branch

### PR DESCRIPTION
**Pull request by krishvoor**
## Description

Add support for AKS, these changes will allow users to run autotune on AKS

Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [ ] Requires DB changes

## How has this been tested?

Please describe the tests that were run to verify your changes and steps to reproduce. Please specify any test configuration required. 

- [ ] New Test X
- [x] Functional testsuite

**Test Configuration**
* Kubernetes clusters tested on: AKS [Azure Kubernetes Service] 

## Checklist :dart:

- [x] Followed coding guidelines
- [x] Comments added
- [ ] Dependent changes merged
- [] Documentation updated
- [ ] Tests added or updated

## Additional information

Include any additional information such as links, test results, screenshots here

### CODE REVIEW

File: scripts/aks-helpers.sh [hunk 1]
```diff
@@ -0,0 +1,212 @@
+#!/bin/bash
+#
+# Copyright (c) 2020, 2021 Red Hat, IBM Corporation and others.
```
Comment (chandrams) [inline scripts/aks-helpers.sh:3]: Please update the copyright year

File: scripts/prometheus_on_aks.sh [hunk 2]
```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+#
+# Copyright (c) 2020, 2021 Red Hat, IBM Corporation and others.
```
Comment (chandrams) [inline scripts/prometheus_on_aks.sh:3]: Please update the copyright year

File: scripts/aks-helpers.sh [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+	echo
+	echo "Deploying Performance Profile objects"
+	${kubectl_cmd} apply -f ${AUTOTUNE_PERF_PROFILE_ROS}
+
+	echo "Info: Deploying autotune yaml to AKS cluster"
+
+	# Replace autotune docker image in deployment yaml
+	sed -e "s|{{ AUTOTUNE_IMAGE }}|${AUTOTUNE_DOCKER_IMAGE}|" ${AUTOTUNE_DEPLOY_MANIFEST_TEMPLATE} > ${AUTOTUNE_DEPLOY_MANIFEST}
+	#sed -i "s|{{ HPO_IMAGE }}|${HPO_DOCKER_IMAGE}|" ${AUTOTUNE_DEPLOY_MANIFEST}
```
Comment (chandrams) [inline scripts/aks-helpers.sh:76]: You can remove this if not required

File: scripts/aks-helpers.sh (hunk 1 above)
Comment (krishvoor) [reply scripts/aks-helpers.sh]: Ack, updated!

File: scripts/prometheus_on_aks.sh (hunk 2 above)
Comment (krishvoor) [reply scripts/prometheus_on_aks.sh]: Ack, updated!

File: scripts/aks-helpers.sh (hunk 3 above)
Comment (krishvoor) [reply scripts/aks-helpers.sh]: Updated, resolving the conversation!

**Review by chandrams — APPROVED** [review]
LGTM!

**Comment by chandrams** [comment]
@krishvoor - Can you sign your commits with a gpg key.

**Comment by dinogun** [comment]
@chandrams I see the the `build_crc` PR check is failing due to an issue with perf profile, is that expected?

**Comment by chandrams** [comment]
@dinogun Some of the pods haven't comeup including the Kruize pod
```
Info: Waiting for kruize to come up.....
Name:             kruize-7f8d4968ff-q7knd
Namespace:        monitoring
Priority:         0
Service Account:  default
Node:             fv-az771-266/10.1.0.39
Start Time:       Thu, 29 Aug 2024 12:25:42 +0000
Labels:           app=kruize
                  name=kruize
                  pod-template-hash=7f8d4968ff
Annotations:      <none>
Status:           Pending
IP:               172.17.0.15
IPs:
  IP:           172.17.0.15
Controlled By:  ReplicaSet/kruize-7f8d4968ff
Init Containers:
  wait-for-kruize-db:
    Container ID:  docker://4db85d233cf23198aab179c62d3e5902e7b23c31401aa2d5711edd8eb3b8aa92
    Image:         quay.io/kruizehub/postgres:15.2
    Image ID:      docker-pullable://quay.io/kruizehub/postgres@sha256:3413592a75ef9f0a664695a0b822287533407222886692b580c4d434444eda11
    Port:          <none>
    Host Port:     <none>
    Command:
      sh
      -c
      until pg_isready -h kruize-db-service -p 5432 -U admin; do
        echo "Waiting for kruize-db-service to be ready..."
        sleep 2
      done
      
    State:          Running
      Started:      Thu, 29 Aug 2024 12:25:54 +0000
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-xrgvk (ro)
Containers:
  kruize:
    Container ID:   
    Image:          autotune_operator:pr_1226
    Image ID:       
    Port:           8080/TCP
    Host Port:      0/TCP
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Environment:
      LOGGING_LEVEL:       info
      ROOT_LOGGING_LEVEL:  error
      DB_CONFIG_FILE:      /etc/config/dbconfigjson
      KRUIZE_CONFIG_FILE:  /etc/config/kruizeconfigjson
      JAVA_TOOL_OPTIONS:   -XX:MaxRAMPercentage=80
    Mounts:
      /etc/config from config-volume (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-xrgvk (ro)
Conditions:
  Type              Status
  Initialized       False 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  config-volume:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      kruizeconfig
    Optional:  false
  default-token-xrgvk:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-xrgvk
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type    Reason     Age   From               Message
  ----    ------     ----  ----               -------
  Normal  Scheduled  13m   default-scheduler  Successfully assigned monitoring/kruize-7f8d4968ff-q7knd to fv-az771-266
  Normal  Pulling    13m   kubelet            Pulling image "quay.io/kruizehub/postgres:15.2"
  Normal  Pulled     13m   kubelet            Successfully pulled image "quay.io/kruizehub/postgres:15.2" in 11.104629374s
  Normal  Created    13m   kubelet            Created container wait-for-kruize-db
  Normal  Started    13m   kubelet            Started container wait-for-kruize-db

Name:             kruize-db-deployment-765bfbdb97-tzmtw
Namespace:        monitoring
Priority:         0
Service Account:  default
Node:             <none>
Labels:           app=kruize-db
                  pod-template-hash=765bfbdb97
Annotations:      <none>
Status:           Pending
IP:               
IPs:              <none>
Controlled By:    ReplicaSet/kruize-db-deployment-765bfbdb97
Containers:
  kruize-db:
    Image:      quay.io/kruizehub/postgres:15.2
    Port:       5432/TCP
    Host Port:  0/TCP
    Environment:
      POSTGRES_PASSWORD:  admin
      POSTGRES_USER:      admin
      POSTGRES_DB:        kruizeDB
    Mounts:
      /var/lib/postgresql/data from kruize-db-storage (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-xrgvk (ro)
Conditions:
  Type           Status
  PodScheduled   False 
Volumes:
  kruize-db-storage:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  kruize-db-pvc
    ReadOnly:   false
  default-token-xrgvk:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-xrgvk
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason            Age                 From               Message
  ----     ------            ----                ----               -------
  Warning  FailedScheduling  13s (x12 over 13m)  default-scheduler  0/1 nodes are available: 1 pod has unbound immediate PersistentVolumeClaims.
```

@krishvoor - Can you please pull the latest changes in mvp_demo & rebase your changes.  Want to check if this resolve the failure.

**Comment by krishvoor** [comment]
@chandrams Rebased.
>@krishvoor - Can you please pull the latest changes in mvp_demo & rebase your changes. Want to check if this resolve the failure.

**Comment by dinogun** [comment]
@krishvoor Thanks for your contribution!

**Pull request merged**